src/SessionSetup/ServiceDeployManager.cpp

```diff
@@ -104,14 +104,23 @@ void DeferToBackgroundThreadAndWait(QObject* context, Func&& func) {
 }  // namespace
 
 template <typename T>
-static outcome::result<T> MapError(outcome::result<T> result, Error new_error) {
-  if (result) {
+static ErrorMessageOr<T> MapError(ErrorMessageOr<T> result, Error new_error) {
+  if (result.has_value()) {
     return result;
-  } else {
-    const auto new_error_code = make_error_code(new_error);
-    ORBIT_ERROR("%s: %s", new_error_code.message().c_str(), result.error().message().c_str());
-    return outcome::failure(new_error_code);
   }
+  const auto new_error_code = make_error_code(new_error);
+  ORBIT_ERROR("%s: %s", new_error_code.message(), result.error().message());
+  return outcome::failure(new_error_code);
+}
+
+template <typename T>
+static ErrorMessageOr<T> MapError(outcome::result<T> result, Error new_error) {
+  if (result.has_value()) {
+    return result.value();
+  }
+  const auto new_error_code = make_error_code(new_error);
+  return ErrorMessage{
```

Comment:
I guess you could also print the error message here, as in the other overload, but I also dont care

Reply:
I mean ORBIT_ERROR

Reply:
I intentionally didn't do that because printing the error message is just a work around because previously we didn't have the option to return the full context in `outcome::result`. No we can do that with `ErrorMessageOr`.